Add OAuth Readme.

wangxinlei opened the pull request:
Add readme for the sample OAuth project.

pvanderknyff reviewed (commented):
Content looks good, though it some of it seems like it'd be a better fit for the documentation than a readme, it's kind of general how to set up oauth and less about this specific sample. If the oauth documentation ends up duplicating some of this readme, it may be better to just link that documentation in the readme so there's only one place we need to keep track of.

Also, can you address the comments in https://github.com/tableau/connector-plugin-sdk/pull/692 in this PR? Just to keep the discussion in one place as much as possible.

Thanks for taking this on!

wangxinlei commented:
> Content looks good, though it some of it seems like it'd be a better fit for the documentation than a readme, it's kind of general how to set up oauth and less about this specific sample. If the oauth documentation ends up duplicating some of this readme, it may be better to just link that documentation in the readme so there's only one place we need to keep track of.
> 
> Also, can you address the comments in #692 in this PR? Just to keep the discussion in one place as much as possible.
> 
> Thanks for taking this on!

Yeah I agree it's a bit more verbose I'll change that.
Oh the code as of right now already has addressed the comments for removing the logs in connectionBuilder.js and mask secrets in oAuthConfig.xml.

rosswbrown commented:
[The snowflake_oauth changes](https://github.com/tableau/connector-plugin-sdk/commit/a5b3f6ca305769aa56a238f2dc355d87f2f39436) should have gone in as a PR.  It looks like the direct merge caused some noise.  I think its ok to keep since there wasn't any overlap in the files.

wangxinlei commented:
> [The snowflake_oauth changes](https://github.com/tableau/connector-plugin-sdk/commit/a5b3f6ca305769aa56a238f2dc355d87f2f39436) should have gone in as a PR. It looks like the direct merge caused some noise. I think its ok to keep since there wasn't any overlap in the files.

Yep I accidentally delete the dev-2021.1 branch yesterday and restore that, I've worked with Patrick to make sure it contains all changes needed. My bad for using the wrong git command that direct push changes in.

pvanderknyff commented:
> > [The snowflake_oauth changes](https://github.com/tableau/connector-plugin-sdk/commit/a5b3f6ca305769aa56a238f2dc355d87f2f39436) should have gone in as a PR. It looks like the direct merge caused some noise. I think its ok to keep since there wasn't any overlap in the files.
> 
> Yep I accidentally delete the dev-2021.1 branch yesterday and restore that, I've worked with Patrick to make sure it contains all changes needed. My bad for using the wrong git command that direct push changes in.

I've made sure the dev-2021.1 branch is protected, to guard against stuff like that in the future.

pvanderknyff commented on samples/scenarios/snowflake_oauth/README.md:
```diff
@@ -0,0 +1,149 @@
+
+# Readme
+
+This sample connector plugin demonstrates how to add OAuth Authentication type to Snowflake datasource.  This sample plugin code has been tested for all the authentication methods listed here with all platforms (Mac Desktop, Windows Desktop and Server, Linux Server). You can use this as an example for how to add OAuth to your own connector. 
+For Under Development connetor, refer to this page on how to enable it locally: https://tableau.github.io/connector-plugin-sdk/docs/share.
+
+## <a id="purpose"/> Use case(s) of this connector plugin
+1. Use **OAuth** authentication to connect to Snowflake with `Tableau Desktop on Windows`/`Tableau Desktop on Mac`. See [OAuth for Tableau Desktop](#desktop)    
+1. Publish a viz from `Tableau Desktop` using **Prompt** and open it in Tableau Server. See [OAuth for Tableau Server](#server)
+1. Publish a viz from `Tableau Desktop` using **Embed [your_username]** and open it in Tableau Server. See [OAuth for Tableau Server](#server)
```
This section is specific to this connector sample, but the rest seems more general. Willing to be pushed back on this, but can we replace it with a link to the documentation so we don't have to worry about keeping this readme up-to-date with future changes?

pvanderknyff reviewed (approved):
Approving for now. I still think this is a bit much for the readme, but as long as this information does not only live here it should be fine, since the kerberos sample already did this style.

pvanderknyff commented on samples/scenarios/snowflake_oauth/README.md:
```diff
@@ -0,0 +1,149 @@
+
+# Readme
+
+This sample connector plugin demonstrates how to add OAuth Authentication type to Snowflake datasource.  This sample plugin code has been tested for all the authentication methods listed here with all platforms (Mac Desktop, Windows Desktop and Server, Linux Server). You can use this as an example for how to add OAuth to your own connector. 
```
```suggestion
This sample connector demonstrates how to add OAuth Authentication type to Snowflake datasource.  This sample connector has been tested for all the authentication methods listed here with all platforms (Mac Desktop, Windows Desktop and Server, Linux Server). You can use this as an example for how to add OAuth to your own connector. 
```
